executor/adapter.go

```diff
@@ -295,6 +296,9 @@ func (a *ExecStmt) RebuildPlan(ctx context.Context) (int64, error) {
 	a.SnapshotTS = ret.LastSnapshotTS
 	a.IsStaleness = ret.IsStaleness
 	a.ReplicaReadScope = ret.ReadReplicaScope
+	if a.Ctx.GetSessionVars().GetReplicaRead().IsClosestRead() && a.ReplicaReadScope == kv.GlobalReplicaScope {
+		logutil.BgLogger().Warn(fmt.Sprintf("tidb can't read closest replicas due to it haven't %s label", placement.DCLabelKey))
```

Comment:
Will this cause a massive amount of logs to be printed if the QPS is high?

Reply:
It can be, but `replica-read` only affect session query, the log won't spread for the whole server.